src/Microsoft.ML.OnnxTransformer/OnnxUtils.cs

```diff
@@ -20,67 +21,135 @@ namespace Microsoft.ML.Transforms.Onnx
+            public List<string> OutputNames { get; }
+            /// <summary>
+            /// Inputs of the containing <see cref="OnnxModel"/>.
+            /// </summary>
+            public OnnxVariableInfo[] InputsInfo { get; }
+            /// <summary>
+            /// Outputs of the containing <see cref="OnnxModel"/>.
+            /// </summary>
+            public OnnxVariableInfo[] OutputsInfo { get; }
 
-            public OnnxModelInfo(IEnumerable<OnnxNodeInfo> inputsInfo, IEnumerable<OnnxNodeInfo> outputsInfo)
+            public OnnxModelInfo(IEnumerable<OnnxVariableInfo> inputsInfo, IEnumerable<OnnxVariableInfo> outputsInfo)
             {
+                InputNames = inputsInfo.Select(val => val.Name).ToList();
                 InputsInfo = inputsInfo.ToArray();
+                OutputNames = outputsInfo.Select(val => val.Name).ToList();
                 OutputsInfo = outputsInfo.ToArray();
             }
+
+            /// <summary>
+            /// Return the ONNX value for a <see cref="IDataView"/> input column called <paramref name="name"/>.
+            /// </summary>
+            public OnnxVariableInfo GetInput(string name)
+            {
+                var index = InputNames.IndexOf(name);
+                if (index < 0)
+                    throw Contracts.ExceptParamValue(name, nameof(name), $"Input tensor, {name}, does not exist in the ONNX model. " +
+                        $"Available input names are [{string.Join(",", InputNames)}].");
+                return InputsInfo[index];
+            }
+
+            /// <summary>
+            /// Return the ONNX value for a <see cref="IDataView"/> output column called <paramref name="name"/>.
+            /// </summary>
+            public OnnxVariableInfo GetOutput(string name)
+            {
+                var index = OutputNames.IndexOf(name);
+                if (index < 0)
+                    throw Contracts.ExceptParamValue(name, nameof(name), $"Onput tensor, {name}, does not exist in the ONNX model. " +
+                        $"Available output names are [{string.Join(",", OutputNames)}].");
+                return OutputsInfo[index];
+            }
         }
 
         /// <summary>
         /// OnnxNodeInfo contains all the information for a given node (e.g. inputs/outputs)
         /// of an Onnx model.
         /// </summary>
-        public class OnnxNodeInfo
+        public class OnnxVariableInfo
         {
             /// <summary>
             /// The Name of the node
             /// </summary>
-            public readonly string Name;
+            public string Name { get; }
             /// <summary>
             /// The shape of the node
             /// </summary>
-            public readonly OnnxShape Shape;
+            public OnnxShape Shape { get; }
             /// <summary>
             /// The type of the node
             /// </summary>
-            public readonly System.Type Type;
+            public System.Type OrtType { get; }
```

Comment:
I don't really like `OrtType` and `MlnetType` for names. How about `OnnxType` and `DataViewType` instead?

Reply:
I will do `TypeInOnnxRuntime` and `DataViewType`. Runtime can freely translate ONNX type to whatever it likes; for example, ONNX tensor can be mapped to `float[]`, `Tensor`, etc.